plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/M2RepositoryService.java

```diff
@@ -15,26 +15,79 @@
 import org.eclipse.core.runtime.Path;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
 
 import com.google.cloud.tools.eclipse.appengine.libraries.MavenCoordinates;
+import com.google.cloud.tools.eclipse.util.MavenUtils;
 
 /**
  * Implementation of {@link ILibraryRepositoryService} that relies on M2Eclipse to download the artifacts and store
  * them in the local Maven repository pointed to by M2Eclipse's M2_REPO variable.
  */
-// FIXME For now this class is just a mock, to be implemented soon
+@Component
 public class M2RepositoryService implements ILibraryRepositoryService {
 
   @Override
-  public IPath getJarLocation(MavenCoordinates mavenCoordinates) {
-    return new Path("/path/to/jar/file/in/m2_repo/" + mavenCoordinates.getArtifactId() + "." + mavenCoordinates.getType());
+  public IPath getJarLocation(MavenCoordinates mavenCoordinates) throws LibraryRepositoryServiceException {
+    try {
+      List<ArtifactRepository> repository = getRepository(mavenCoordinates);
+
+      Artifact artifact;
+      if (mavenCoordinates.getVersion().equals(MavenCoordinates.LATEST_VERSION)) {
+        artifact = MavenUtils.resolveLatestReleasedArtifact(null, mavenCoordinates.getGroupId(), mavenCoordinates.getArtifactId(), mavenCoordinates.getType(), mavenCoordinates.getClassifier(), repository);
+      } else {
+        artifact = MavenUtils.resolveArtifact(null, mavenCoordinates.getGroupId(), mavenCoordinates.getArtifactId(), mavenCoordinates.getType(), mavenCoordinates.getVersion(), mavenCoordinates.getClassifier(), repository);
+      }
+
+      return new Path(artifact.getFile().getAbsolutePath());
+    } catch (CoreException ex) {
+      throw new LibraryRepositoryServiceException("Could not resolve maven artifact: " + mavenCoordinates, ex);
+    }
+  }
+
+  private ArtifactRepository getRepository(String repository) throws LibraryRepositoryServiceException {
+    try {
+      URI repoUri = new URI(repository);
+      if (!repoUri.isAbsolute()) {
+        throw new LibraryRepositoryServiceException("repository URI must be an absolute URI (i.e. has to have a "
+            + "schema component)");
+      }
+      return MavenPlugin.getMaven().createArtifactRepository(repoUri.getHost(), repoUri.toString());
+    } catch (URISyntaxException exception) {
+      throw new LibraryRepositoryServiceException("repository is not a valid URI and currenlty only 'central' is "
+          + "supported as repository ID",
+          exception);
+    } catch (CoreException exception) {
+      throw new LibraryRepositoryServiceException("Could not create remote repository: " + repository, exception);
+    }
+  }
+
+  private List<ArtifactRepository> getRepository(MavenCoordinates mavenCoordinates) throws LibraryRepositoryServiceException {
+    if (mavenCoordinates.getRepository().equals(MavenCoordinates.MAVEN_CENTRAL_REPO)) {
```

Comment:
If you put the constant first you protect against some NullPointerExceptions. E.g.

MavenCoordinates.MAVEN_CENTRAL_REPO.equals(mavenCoordinates.getRepository())